chain/network/src/peer.rs

```diff
@@ -731,6 +743,20 @@ impl StreamHandler<Result<Vec<u8>, ReasonForBan>> for Peer {
                 return;
             }
         };
+        // Drop duplicated messages routed within 1s
+        if let PeerMessage::Routed(msg) = &peer_msg {
+            let key = (msg.author.clone(), msg.target.clone(), msg.signature.clone());
+            let now = Utc::now();
```

Comment:
Separately from the calendar-clock vs monotonic clock issue, it feels a bit odd that we just reach out for *current* time in the middle of computation. This makes such code difficult to test, as you can't trivially change the time. I would probably expect the timestamp of receiving to be just a part of the message, and used throughout the code. 

But this is just a chache, so it's also not a big deal if we are not strict about time.

Reply:
@matklad Currently, the timestamp is not a part of the message. We can't change the API.

However, we can change our code to use a function to get time, instead of `Instant::now()`. 
For unit tests, we can have api that mocks this function, and changes it's behavior.